funcx_endpoint/funcx_endpoint/executors/high_throughput/funcx_worker.py

```diff
@@ -92,6 +94,11 @@ def __init__(self, worker_id, address, port, logdir, debug=False, worker_type='R
         logger.info('Trying to connect to : tcp://{}:{}'.format(self.address, self.port))
         self.task_socket.connect('tcp://{}:{}'.format(self.address, self.port))
         self.poller.register(self.task_socket, zmq.POLLIN)
+        signal.signal(signal.SIGTERM, self.handler)
+
+    def handler(self, signum, frame):
+        logger.error("Signal handler called with signal", signum)
+        raise TaskCancelled(0, 0)
```

Comment:
these 0,0 constants are suspicious - if it's a different kind of exception to an exception with meaningful values there, make it a different class